pkg/nsx/services/firewall.go

```diff
@@ -592,3 +593,69 @@ func buildPeerTags(obj *v1alpha1.SecurityPolicy, peers *[]v1alpha1.SecurityPolic
 	}
 	return peerTags
 }
+
+func (service *SecurityPolicyService) deleteGroup(groupsClient domains.GroupsClient, nsxGroup *model.Group) error {
+	failIfSubtreeExistsParam := false
+	forceParam := false
+	return groupsClient.Delete(getDomain(), *nsxGroup.Id, &failIfSubtreeExistsParam, &forceParam)
+}
+
+func (service *SecurityPolicyService) DeleteSecurityPolicy(key types.UID) error {
+	groupsClient := service.NSXClient.GroupClient
+	policiesClient := service.NSXClient.SecurityClient
+
+	policy, err := service.SecurityPolicyStore.ByIndex(util.TagScopeSecurityPolicyCRUID, string(key))
+	if err != nil {
+		log.Error(err, "failed to find SecurityPolicy", "key", string(key))
+		return err
+	}
+	nsxSecurityPolicy := policy[0].(model.SecurityPolicy)
+	err = policiesClient.Delete(getDomain(), *nsxSecurityPolicy.Id)
+	if err != nil {
+		log.Error(err, "failed to delete SecurityPolicy", "key", string(key))
+		return err
+	}
+	service.SecurityPolicyStore.Delete(nsxSecurityPolicy)
+
+	rules, err := service.SecurityPolicyStore.ByIndex(util.TagScopeSecurityPolicyCRUID, string(key))
+	if err != nil {
+		log.Error(err, "failed to find Rules", "key", string(key))
```

Comment:
replace "key" to "UID" or "ID" based on the meaning

Reply:
done